### PR TITLE
Provide RepositoryBranch metadata for package

### LINKED ISSUE #23: Provide RepositoryBranch metadata for package, just like the other metadata
As documented [in nuget](https://learn.microsoft.com/en-us/nuget/reference/msbuild-targets) you can provide 
not only the repo URL, type and commit, but also the branch. But this information is not populated by default 
by the SourceLink package, neither the ReproducibleBuilds (unless [this PR is merged](https://github.com/dotnet/reproducible-builds/pull/27)).

So just bring the code in that PR here until that's merged and shipped.

### PR DESCRIPTION
**Pull request by kzu**
As documented in nuget (https://learn.microsoft.com/en-us/nuget/reference/msbuild-targets) you can provide not only the repo URL, type and commit, but also the branch. But this information is not populated by default by the SourceLink package, neither the ReproducibleBuilds (unless the PR at https://github.com/dotnet/reproducible-builds/pull/27 is merged).

So just bring the code in that PR here until that's merged and shipped.

Fixes #23

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -85,7 +85,38 @@
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(RepositoryBranch)' == ''">
```
Comment (Sergio0694) [inline Directory.Build.props:88]: This is a lot of code and I don't really it's worth adding it, especially given the fix is already pending in the official package. I'd rather just wait for that to be available, and then we can just update this PR to bump the dependency 🙂

Comment (kzu) [reply Directory.Build.props:88]: Yeah, that might be best. Also, if you only build on GH actions, might be worth keeping just that single part?

Comment (Sergio0694) [reply Directory.Build.props:88]: Maybe, but given you've already opened [a PR in ther](https://github.com/dotnet/reproducible-builds/pull/27) and it also already has an approval, I'd rather just wait for that one to be in so we can do this properly from the start. I can leave this open if you want, so you can then just rebase and bump that dependency 🙂

**Comment by Sergio0694** [comment]
Closing, since the same fix has been merged into https://github.com/dotnet/reproducible-builds, so it'll propagate here whenever they push a new release that we can also reference. I'll update and publish a new version of PolySharp when that happens.

Thank you for contributing that upstream! 🙂

**Comment by kzu** [comment]
Hey @Sergio0694, TBH that project seems to be pretty much dead. No releases since Nov-2021 and nobody is actively answering anything there. Perhaps go the other way and take the change here and delete once they release?

**Pull request closed without merging**